Alternative partial derivative calculation for interpolation

motloch opened the pull request:
Major change is allowing the user to choose alternative way how to calculate the partial derivatives for interpolation (less precise / faster than original toms760).

Further minor changes:
* Random number generator now labeled as initialized even when using seed
* Makes sure results are sensible even if input unlensed cmb has CTT*CEE < CTE**2
* Moved interpolation x, y to double precision to avoid certain map artifacts
* Increased interp_edge to have correct behavior around the edges

cmbant commented:
Thanks!
What is "if (Examp .ne. Examp)" supposed to do? It might also be a good idea to add your new parameter to the example .ini files (set to the old default, but documenting the new option -  I think the default .ini will fail as the new option is currently not defined?).

motloch commented:
It should check for NaN. Rewrote the condition so it is clearer what it is checking on.

Good catch, updated example "params.ini" so now it can run as provided.

cmbant commented:
Looks good, thanks.
